presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -154,13 +154,17 @@
     // for materialized view analysis state detection, state is used to identify if materialized view has been expanded or in-process.
     private final Map<Table, MaterializedViewAnalysisState> materializedViewAnalysisStateMap = new HashMap<>();
 
+    // for recursive base query detection
+    private boolean isStatementRewriteWithOptimizationEnabled;
+
     public Analysis(@Nullable Statement root, List<Expression> parameters, boolean isDescribe)
     {
         requireNonNull(parameters);
 
         this.root = root;
         this.parameters = ImmutableList.copyOf(requireNonNull(parameters, "parameters is null"));
         this.isDescribe = isDescribe;
+        this.isStatementRewriteWithOptimizationEnabled = true;
```

Comment:
I think this flag isStatementRewriteWithOptimizationEnabed is related to the session property query_optimization_with_materialized_view_enabled. Is it? If so, are you able to initialize isStatementRewriteWithOptimizationEnabled here according to session property query_optimization_with_materialized_view_enabled? Or better way is to check the session property on the fly wherever you need to check this flag now

Reply:
The query_optimization_with_materialized_view_enabled is for the session configuration. This boolean (which is now renamed to isQueryOptimizedUsingMaterializedView) is used to test if the current statement has already been optimized using materialized, which avoid further unnecessary rewrite.